sh_scrapy/settings.py

```diff
@@ -96,13 +95,25 @@ def _load_addons(addons, settings, merged_settings, priority=0):
         skey = _get_component_base(settings, addon['type'])
         components = settings[skey]
         path = update_classpath(addon_path)
-        components[path] = addon['order']
+        if path not in components:
```

Comment:
If I'm not mistaken components dict can contain deprecated paths because `update_classpath` is used only when in [build_component_list](https://github.com/scrapy/scrapy/blob/ebef6d7c6dd8922210db8a4a44f48fe27ee0cd16/scrapy/utils/conf.py#L13) so we need to check if `components` contains any of `path` or `addon_path`.